packages/modules-sdk/src/medusa-app.ts

```diff
@@ -0,0 +1,179 @@
+import { RemoteFetchDataCallback } from "@medusajs/orchestration"
+import {
+  LoadedModule,
+  MODULE_RESOURCE_TYPE,
+  MODULE_SCOPE,
+  ModuleConfig,
+  ModuleJoinerConfig,
+  ModuleServiceInitializeOptions,
+  RemoteJoinerQuery,
+} from "@medusajs/types"
+import {
+  ContainerRegistrationKeys,
+  ModulesSdkUtils,
+  isObject,
+} from "@medusajs/utils"
+import { MODULE_PACKAGE_NAMES, Modules } from "./definitions"
+import { MedusaModule } from "./medusa-module"
+import { RemoteLink } from "./remote-link"
+import { RemoteQuery } from "./remote-query"
+
+export type MedusaModuleConfig = (Partial<ModuleConfig> | Modules)[]
+export type SharedResources = {
+  database?: ModuleServiceInitializeOptions["database"] & {
+    pool?: {
+      min?: number
+      max?: number
+      idleTimeoutMillis?: number
+      idleInTransactionSessionTimeout?: number
+    }
+  }
+}
+
+export async function MedusaApp({
+  sharedResourcesConfig,
+  modulesConfigPath,
+  modulesConfig,
+  linkModules,
+  remoteFetchData,
+}: {
+  sharedResourcesConfig?: SharedResources
+  loadedModules?: LoadedModule[]
+  modulesConfigPath?: string
+  modulesConfig?: MedusaModuleConfig
+  linkModules?: ModuleJoinerConfig | ModuleJoinerConfig[]
+  remoteFetchData?: RemoteFetchDataCallback
+} = {}): Promise<{
+  modules: Record<string, LoadedModule | LoadedModule[]>
+  link: RemoteLink | undefined
+  query: (
+    query: string | RemoteJoinerQuery,
+    variables?: Record<string, unknown>
+  ) => Promise<any>
+}> {
+  const modules: MedusaModuleConfig =
+    modulesConfig ??
+    (await import(process.cwd() + (modulesConfigPath ?? "/modules-config")))
+      .default
+
+  const injectedDependencies: any = {}
+
+  const dbData = ModulesSdkUtils.loadDatabaseConfig(
+    "medusa",
+    sharedResourcesConfig as ModuleServiceInitializeOptions,
+    true
+  )!
+  const { pool } = sharedResourcesConfig?.database ?? {}
+
+  if (dbData?.clientUrl) {
+    const { knex } = await import("@mikro-orm/knex")
```

Comment:
thought: Should this be pure `knex` as with the loader in the core?

Reply:
That could be it. I will replace it.
But soon the module loader from `/medusa` will use this `MedusaApp` as well